src/time.c

```diff
@@ -46,6 +46,61 @@ rcutils_time_point_value_as_nanoseconds_string(
   return RCUTILS_RET_OK;
 }
 
+rcutils_ret_t
+rcutils_time_point_value_as_date_string(
+  const rcutils_time_point_value_t * time_point,
+  char * str,
+  size_t str_size)
+{
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(time_point, RCUTILS_RET_INVALID_ARGUMENT);
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(str, RCUTILS_RET_INVALID_ARGUMENT);
+  if (0 == str_size) {
+    return RCUTILS_RET_OK;
+  }
+  // best to abs it to avoid issues with negative values in C89, see:
+  //   https://stackoverflow.com/a/3604984/671658
+  uint64_t abs_time_point = (uint64_t)llabs(*time_point);
+  // break into two parts to avoid floating point error
+  uint64_t seconds = abs_time_point / (1000u * 1000u * 1000u);
+  uint64_t nanoseconds = abs_time_point % (1000u * 1000u * 1000u);
+  // Make sure the buffer is large enough to hold the largest possible uint64_t
+  char nanoseconds_str[21];
+
+  if (rcutils_snprintf(nanoseconds_str, sizeof(nanoseconds_str), "%" PRIu64, nanoseconds)) {
```

Comment:
This check isn't correct.  The [documentation for rcutils_snprintf](https://docs.ros.org/en/rolling/p/rcutils/generated/function_snprintf_8h_1aa3d934f197ccb63c08052a27192478ec.html#function-rcutils-snprintf) says:

```
Returns:
    the number of bytes that would have been written given enough space, or
Returns:
    a negative number if there is an error, but unlike _snprintf_s(), -1 is not returned if there is truncation.
```

So we should check that this is > 0.

Reply:
shouldn't we check if the return value is -1 or 0?

Reply:
Yes, you are right.  We can change the check to `<=0`.